src/htr2hpc/api_client.py

```diff
@@ -0,0 +1,317 @@
+            print("Error: not found")
+
+        if resp.status_code == requests.codes.unauthorized:
+            print("Error: not authorized")
+
+        # TODO: error handling / logging for other cases?
+        # (useful for dev if nothing else...)
+
+    def models(self):
+        """paginated list of models"""
+        api_url = "models/"
+        resp = self._make_request(api_url)
+        return ResultsList(result_type="model", **resp.json())
+
+    def update_model(self, model_id: int, model_file: pathlib.Path):
+        """Update an existing model record with a new model file."""
+        api_url = f"models/{model_id}/"
+        with open(model_file, "rb") as mfile:
+            files = {"file": mfile}
+            data = {
+                "name": "updated model",
+                "job": "Segment",  # required; get from existing record?
+                # file_size (int)  - set from pathlib object using .stat().st_size
+                # versions ?
+                # accuracy_percent
+            }
+            resp = self._make_request(api_url, method="PUT", files=files, data=data)
+        # on successful update, returns the model object
+        return to_namedtuple("model", resp.json())
+
+    def documents(self):
+        """paginated list of documents"""
+        api_url = "documents/"
+        resp = self._make_request(api_url)
+        return ResultsList(result_type="document", **resp.json())
+
+    def document(self, document_id: int):
+        """details for a single document"""
+        api_url = f"documents/{document_id}/"
+        resp = self._make_request(api_url)
+        return to_namedtuple("document", resp.json())
+
+    def document_export(
+        self, document_id: int, transcription_id: int, include_images: bool = False
+    ):
+        """request a document export be compiled for download"""
+        api_url = f"documents/{document_id}/export/"
+        # export form requires a region_types list, which
+        # is a based on block type ids for this document
+        # and may also include "Undefined" and "Orphan"
+        document_info = self.document(document_id)
+        block_types = [block.pk for block in document_info.valid_block_types]
+        block_types.extend(["Undefined", "Orphan"])
+
+        data = {
+            "transcription": transcription_id,  # could be multiple(?)
+            "file_format": "alto",
+            "include_images": True,
+            "region_types": block_types,
+        }
+
+        resp = self._make_request(api_url, method="POST", data=data)
+        return to_namedtuple("status", resp.json())
+
+    def export_file_url(
+        self,
+        user_id: int,
+        document_id: int,
+        document_name: str,
+        file_format: str,
+        creation_time: datetime.datetime,
+    ) -> str:
+        """Generate URL for an export file generated by :meth:`document_export`"""
+        # is there any way get current user id from api based purely on token?
```

Comment:
> `# is there any way get current user id from api based purely on token?`

Yes, the following API URL returns the current user only: https://test-htr.lib.princeton.edu/api/users/current/

Reply:
Ah, thank you! How did you know that? Is there other documentation / code I should be looking at to understand the API better?

Reply:
Yes, I was viewing https://test-htr.lib.princeton.edu/api/users/ in the browser (mostly to check whether I was right to think that regular users would only ever see themselves, but admins would see everyone) when I noticed `Current` under the Extra Actions. So unfortunately no extra documentation I can point you to, but there are some helpful methods we can keep an eye out for!

Reply:
Ooh! I hadn't noticed that extra actions yet! That is good to know about.

I see now that there's a documents tasks api endpoint, maybe it would be better to use that one for getting the export status...